### PR TITLE
[ACR] Update to 2017-06-01-preview version

### PR DESCRIPTION
**Pull request by djyou**
Add webhooks and replications as nested resources under registries.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by djyou** [comment]
@DavidObando Rebased from master branch with managed registry, webhooks, and replications.
@SajayAntony

**Comment by djyou** [comment]
Please ignore all the SDK files, they will be removed before sending a PR to Azure repo.

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/replication.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import azure.cli.core.azlogging as azlogging
+logger = azlogging.get_az_logger(__name__)
+
+def acr_replication_list(registry_name,
+                         resource_group_name=None):
+    '''Lists all the replications for the specified container registry.
+    :param str registry_name: The name of container registry
+    '''
+    if resource_group_name is None:
+        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
```
Comment (SajayAntony) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/replication.py]: Can you "functionize" - resource_group_name  = get_resource_group_name(registry, resource_group_name)   and do the If check only in that function?

File: src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py [hunk 2]
```diff
@@ -19,3 +20,29 @@ def validate_registry_name(namespace):
...
+        result = {comps[0]: comps[1]} if len(comps) > 1 else {string: ''}
+    return result
+
+def validate_replication_location(namespace):
+    if namespace.location:
+        registry_location, _ = get_registry_location_by_name(
+            namespace.registry_name, namespace.resource_group_name)
+        if namespace.location == "".join(registry_location.split()).lower():
+            raise CLIError(
+                "Replication should not be in the same location '{}' as the registry."
```
Comment (SajayAntony) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py]: >"Replication should not be in the same location '{}' as the registry." [](start = 16, length = 70)

When adding another region don't we make sure that the current region is also there in the replication list?

Comment (djyou) [inline src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py]: I might end up removing this validation. The difference is whether we want to fail the request quickly on the client side or send the request to RP and then fail.

**Pull request merged**